pkg/replication/rollout_order.go

```diff
@@ -88,19 +80,19 @@ func (r *rolloutOrder) compareErrors(iErr, jErr error) compare {
 }
 
 func (r *rolloutOrder) compareHealth(iNode, jNode allocation.Node, iHealth, jHealth *health.ServiceNodeStatus) compare {
-	if iHealth.Healthy && !jHealth.Healthy {
-		return greaterThan
-	}
-	if iHealth.Healthy && jHealth.Healthy {
-		return compareFromBool(iNode.Name < jNode.Name)
-	}
-	if !iHealth.Healthy && jHealth.Healthy {
-		return lessThan
-	}
-	if !iHealth.Healthy && !jHealth.Healthy {
-		return compareFromBool(iNode.Name < jNode.Name)
+	comp := health.Compare(iHealth.Health, jHealth.Health)
+	if comp == 0 {
```

Comment:
That will fail because `equal` is of type compare and `comp` is of type int. I guess you can cast it. Personally I don't see the value in having an extra type just for these three constant integers.